cscs-checks/system/io/ior_check.py

```diff
@@ -1,160 +1,165 @@
-            self.num_tasks_per_node = 12
-        elif fs_mount_point == '/scratch/snx2000':
-            self.valid_systems = ['dom:gpu']
-            if self.current_system.name == 'dom':
-                self.num_tasks = 192
-                self.num_tasks_per_node = 12
-            else:
-                self.num_tasks = 1
-                self.num_tasks_per_node = 1
-        elif fs_mount_point == '/scratch/snx2000tds':
-            self.valid_systems = ['dom:gpu']
-            if self.current_system.name == 'dom':
-                self.num_tasks = 192
-                self.num_tasks_per_node = 12
-            else:
-                self.num_tasks = 1
-                self.num_tasks_per_node = 1
-        elif fs_mount_point == '/scratch/snx3000':
-            self.valid_systems = ['daint:gpu']
-            self.num_tasks = 720
-            self.num_tasks_per_node = 12
-        elif fs_mount_point == '/users':
-            self.valid_systems = ['daint:gpu', 'dom:gpu']
-            self.num_tasks = 1
-            self.num_tasks_per_node = 1
-            self.tags |= {'maintenance'}
-        elif fs_mount_point == '/apps':
-            self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.descr = 'IOR check (%s)' % base_dir
+        self.tags = {'ops', base_dir}
+
+        self.base_dir = base_dir
+        self.username = getpass.getuser()
+        self.test_dir = os.path.join(self.base_dir, self.username, '.ior')
+        self.test_file = os.path.join(self.test_dir, 'ior.dat')
+
+        try:
+            os.mkdir(self.test_dir)
+        except OSError:
+            pass
+
+        self.valid_systems = fs[base_dir]['valid_systems']
+
+        cur_sys = self.current_system.name
+
+        try:
+            self.num_tasks = fs[base_dir][cur_sys]['num_tasks']
+        except KeyError:
             self.num_tasks = 1
+        try:
+            tasks_per_node = fs[base_dir][cur_sys]['num_tasks_per_node']
+            self.num_tasks_per_node = tasks_per_node
+        except KeyError:
             self.num_tasks_per_node = 1
 
-        self.valid_prog_environs = ['PrgEnv-cray']
+        self.ior_block_size = fs[base_dir]['ior_block_size']
+        self.ior_access_type = fs[base_dir]['ior_access_type']
+        self.executable_opts = ['-B', '-F', '-C ', '-Q 1', '-t 4m', '-D 30',
+                                '-b', self.ior_block_size,
+                                '-a', self.ior_access_type,
+                                '-o', self.test_file]
         self.sourcesdir = os.path.join(self.current_system.resourcesdir, 'IOR')
         self.executable = os.path.join('src', 'C', 'IOR')
         self.build_system = 'Make'
+
+        try:
+            prog_env = fs[base_dir][cur_sys]['valid_prog_environs']
+            self.valid_prog_environs = prog_env
+        except KeyError:
+            self.valid_prog_environs = ['PrgEnv-cray']
+
+        try:
+            self.build_system.cc = fs[base_dir][cur_sys]['build_system_cc']
```

Comment:
Check my comment above. You won't need to set this if Fulen is properly configured.